Lib/getpass.py

```diff
@@ -151,6 +175,33 @@ def _raw_input(prompt="", stream=None, input=None):
     return line
 
 
+def _input_with_echochar(prompt, stream, input, echochar):
+    if not stream:
+        stream = sys.stderr
+    if not input:
+        input = sys.stdin
+    prompt = str(prompt)
+    stream.write(prompt)
+    stream.flush()
+    passwd = ""
+    while True:
+        char = input.read(1)
+        if char == '\n' or char == '\r':
+            break
+        if char == '\x03':
```

Comment:
how should we handle `^D`? and/or nullbytes?

Reply:
I checked what the behavior is if you don't pass `echochar` and there you need to double-click `^D` to trigger the input interrupt. so I made the same behavior.

about nullbytes I don't know how to reproduce it so I decided to ignore such input

Reply:
It could be from another stream. Here, input is an arbitrary stream. Mocking streams, receiving from connection or just finding embedded null bytes from somewhere.

Reply:
Got it, but is it okay to ignore it?